### PR TITLE
#1725 Version Release Notes Generation

### PR DESCRIPTION
**Pull request by ldraney**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

The goal of this PR is to create a draft release and release notes as a part of our pipeline, as well as providing a convenient link for QA to review the draft release notes.  

The release notes are generated with the same file our current release notes are generated from.  They should look [something like this](https://github.com/department-of-veterans-affairs/notification-api/releases/tag/v1.7.9), and should always be generated based off comparing the previous release tag's value with the new release version. 

The javascript files in this PR were formatted according to [prettier](https://prettier.io/docs/en/cli.html)

issue #1725 

## Type of change

Please check the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

### Additional updates
- `getReleaseVersionValue` was moved to actionUtils.js because it is being used in more that just prData.js.  This caused some refactoring of imports, including for `appendSummary()` now needing the `core` parameter. 

## How Has This Been Tested?

How has this been tested? (e.g. Unit tests, Tested locally, Tested as a GitHub action, Depoyed to dev, etc)  
Please provide relevant information and / or links to demonstrate the functionality or fix. (e.g. screenshots, link to deployment, regression test results, etc)

[POC run with Cris here](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/9102452460)

Because the repo variable RELEASE_VERSION currently and accurately reflects the latest merge to release branch, the final test will need to be actually merging this PR.   The above test commented out tag creation and updating the RELEASE_VERSION, as well as the deploys that are not affected by the code changes in this PR. 

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

File: .github/scripts/actionUtils.js [hunk 1]
```diff
@@ -10,7 +13,7 @@ const fs = require("fs"); // NodeJs module provides an API for interacting with
  *                          or rejects if an error occurs during the append operation.
  *
  */
-async function appendSummary(summaryContent) {
+async function appendSummary(core, summaryContent) {
```
Comment (kalbfled) [inline .github/scripts/actionUtils.js:16]: Please update the docstring "params" to reflect the new parameter.

File: .github/workflows/create-and-post-tag.yml [hunk 2]
```diff
@@ -2,15 +2,23 @@ name: Create And Post Tag with Summary
 
 on: 
   workflow_call:
+    outputs:
+      previous_version:
+        description: "Will be used for generating release notes from the previous tag"
```
Comment (cris-oddball) [inline .github/workflows/create-and-post-tag.yml:7]: Present tense please
```suggestion
        description: "Generates release notes from comparing the previous tag to proposed tag"
```

File: .github/workflows/cd-pipeline.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ jobs:
           echo "Deploying commit SHA ${{ github.sha }}, the latest merge to master"
           echo "Deployment paused for manual approval."
 
-# this job deploys the SHA that triggered this workflow
+# # this job deploys the SHA that triggered this workflow
```
Comment (kalbfled) [inline .github/workflows/cd-pipeline.yml:44]: Typo?  I think you just want to increase the indentation.

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: .github/workflows/cd-pipeline.yml [hunk 4]
```diff
@@ -89,14 +89,21 @@ jobs:
           
       - name: Merge commit SHA to release
         run: |
-          git merge ${{ github.sha }} -X theirs
+          git merge ${{ github.sha }} --no-squash -X theirs
```
Comment (kalbfled) [inline .github/workflows/cd-pipeline.yml:92]: My understanding is that this is in lieu of reverting the previous merge.

File: .github/workflows/create-release-notes.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+      - name: Create Release Notes
+        uses: actions/github-script@v7
+        env:
+          previousVersion: ${{ inputs.previous_version }}
+        with:
+          github-token: ${{ secrets.VAR_PAT }}
+          script: |
+            const path = '.github/scripts/createReleaseNotes.js';
+            const createReleaseNotes = require(path);
```
Comment (kalbfled) [inline .github/workflows/create-release-notes.yml:26]: You've done this here and elsewhere--created a variable that is only used once.  It doesn't hurt, but I would just make it one line using the literal.

File: .github/scripts/createReleaseNotes.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      draft: true,
+      prerelease: false,
+    });
+
+    const releaseUrl = response.data.html_url; // Extract URL from the response object
+    console.log("Release URL:", releaseUrl); // Log URL to the console
+    // console.log('Release created successfully:', response);
+    return releaseUrl; // Return the URL
+  } catch (error) {
+    console.error("Error creating release:", error);
```
Comment (kalbfled) [inline .github/scripts/createReleaseNotes.js:42]: Does this end up in the action's logs?

File: .github/scripts/createReleaseNotes.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      name: `${tag_name} - ${formatDate()}`,
+      body,
+      draft: true,
+      prerelease: false,
+    });
+
+    const releaseUrl = response.data.html_url; // Extract URL from the response object
+    console.log("Release URL:", releaseUrl); // Log URL to the console
+    // console.log('Release created successfully:', response);
+    return releaseUrl; // Return the URL
```
Comment (kalbfled) [inline .github/scripts/createReleaseNotes.js:40]: These are not helpful comments.
```suggestion
    const releaseUrl = response.data.html_url;
    console.log("Release URL:", releaseUrl);
    return releaseUrl;
```

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: .github/workflows/cd-pipeline.yml (hunk 4 above)
Comment (ldraney) [reply .github/workflows/cd-pipeline.yml:92]: We did not revert the merge.  We are just readding this flag here.

Comment (ldraney) [reply .github/workflows/cd-pipeline.yml:92]: That is correct

File: .github/workflows/create-and-post-tag.yml (hunk 2 above)
Comment (ldraney) [reply .github/workflows/create-and-post-tag.yml:7]: Thanks for this.  I updated it for the value:
"This value is used for generating release notes from the previous tag"

File: .github/workflows/create-release-notes.yml (hunk 5 above)
Comment (ldraney) [reply .github/workflows/create-release-notes.yml:26]: done, will see on next commit

File: .github/scripts/createReleaseNotes.js (hunk 7 above)
Comment (ldraney) [reply .github/scripts/createReleaseNotes.js:40]: thank you

File: .github/scripts/createReleaseNotes.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+ * Creates a draft release on GitHub with a specified tag and release notes.
+ *
+ * @param {Object} github The authenticated GitHub API client instance.
+ * @param {string} owner The username or organization name on GitHub that owns the repository.
+ * @param {string} repo The repository name.
+ * @param {string} tag_name The tag associated with the release.
+ * @param {string} body The content of the release notes.
+ * @returns {Promise<string>} A Promise that resolves with the URL of the newly created draft release.
+ */
+async function createDraftRelease(github, owner, repo, tag_name, body) {
```
[the remark this answers is not in the document]
Comment (ldraney) [reply .github/scripts/createReleaseNotes.js:25]: All of these files have been formatted with prettier.  

I tried adding the trialing comma, but it actually removes it and reformats it to as its seen here. 

Will you please be more specific?  I believe this is consistent with how other functions are created, for example in prData.js.

[the remark this answers is not in the document]
Comment (cris-oddball) [reply .github/scripts/createReleaseNotes.js:25]: This shouldn't hold up this ticket, resolving convo

**Review by cris-oddball — APPROVED** [review]

**Pull request merged**